src/common/marlin_server.cpp

```diff
@@ -480,16 +480,24 @@ void marlin_server_print_resume(void) {
 }
 
 void marlin_server_print_reheat_start(void) {
-    if ((marlin_server.print_state == mpsPaused) && (marlin_server_print_reheat_ready() == 0)) {
+    if ((marlin_server.print_state == mpsPaused) && marlin_server_print_reheat_ready()) {
         thermalManager.setTargetHotend(marlin_server.resume_nozzle_temp, 0);
     }
 }
 
-int marlin_server_print_reheat_ready(void) {
-    if (marlin_server.vars.target_nozzle == marlin_server.resume_nozzle_temp)
-        if (marlin_server.vars.temp_nozzle >= (marlin_server.vars.target_nozzle - 5))
-            return 1;
-    return 0;
+// Fast temperature recheck.
+// Does not check stability of the temperature.
+bool marlin_server_print_reheat_ready() {
+    // check nozzle
+    if (marlin_server.vars.target_nozzle != marlin_server.resume_nozzle_temp
+        || marlin_server.vars.temp_nozzle < (marlin_server.vars.target_nozzle - TEMP_HYSTERESIS)) {
+        return false;
+    }
+    // check bed
+    if (marlin_server.vars.temp_bed < (marlin_server.vars.target_bed - TEMP_BED_HYSTERESIS))
+        return false;
```

Comment:
• I know it was here to begin with, but that `!=` seems extra (it could be incorporated as `<=` in the main condition).
• It's a bit odd that these checks are slightly different from each other.
• Does it have to go over the temperature, shouldn't the hysteresis work symmetrically to both sides?

Reply:
1. There are 3 variables:
1.1. target_nozzle
1.1. temp_nozzle
1.1. resume_nozzle_temp
They can't be mixed.
1. Looks odd but nozzle can be turned off so we need to set the printing temperature back. No need for the bed.
2. Mathematically should. But there's no issue if bed has higher temperature. There are drawbacks if nozzle has higher temp. but it cools down very quickly once the nozzle is moving and extruding and/or with fan on. Heating is slow so we need to wait.